### PR TITLE
chore(rust): fix clippy lints

### PR DESCRIPTION
**Pull request by niklaslong**
I've only addressed the benign lints not affecting the APIs for now (on `nightly 1.55.0`). 

I'v also preemptively opened my [CLA agreement PR](https://github.com/ockam-network/contributors/pull/24).

### CODE REVIEW

File: implementations/rust/ockam/ockam_entity/src/entity.rs [hunk 1]
```diff
@@ -105,7 +105,7 @@ impl Entity {
 impl ProfileAdd for Entity {
     fn add_profile(&mut self, profile: ProfileSync) -> Result<()> {
         if let Ok(id) = profile.identifier() {
-            if let Some(_) = self.profiles.insert(id, profile) {
+            if self.profiles.insert(id, profile).is_some() {
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_entity/src/entity.rs:108]: @jared-s does it conflict with your changes?

Comment (jared-s) [reply implementations/rust/ockam/ockam_entity/src/entity.rs:108]: oh yes :-) my changeset around this is huge. i will need to take the merge.

**Comment by SanjoDeundiak** [comment]
@niklaslong thank you for the contribution! LGTM
@jared-s let's proceed to merging it

**Review by SanjoDeundiak — APPROVED** [review]

**Comment by jared-s** [comment]
Merged as https://github.com/ockam-network/ockam/pull/1501 - Thank you!

**Pull request closed without merging**